tests/unit/mocked/replicator_test.py

```diff
@@ -127,34 +142,52 @@ def test_create_replication_errors(self):
             )
 
     def test_list_replications(self):
-        with mock.patch('cloudant.database.CloudantDatabase.all_docs') as mock_all_docs:
+        """ test retrieve replications"""
+        def getitem(name):
+            db_dict = {'_replicator': CouchDatabase(
+                mock_account,
+                '_replicator'
+            )}
+            return db_dict[name]
+        with mock.patch('cloudant.database.CouchDatabase.all_docs') as mock_all_docs:
             mock_all_docs.return_value = {
                 "rows": [
-                    {"doc":"replication_1"},
-                    {"doc": "replication_2"}
+                    {"id": "replication_1", "doc": {"_id": "replication_1"}},
+                    {"id": "replication_2", "doc": {"_id": "replication_2"}}
                 ]
             }
             mock_account = mock.Mock()
-            repl = ReplicatorDatabase(mock_account)
+            mock_account.__getitem__ = mock.Mock()
+            mock_account.__getitem__.side_effect = getitem
+            repl = Replicator(mock_account)
 
             self.assertEqual(
                 repl.list_replications(),
-                ['replication_1', 'replication_2']
+                [{"_id": "replication_1"}, {"_id": "replication_2"}]
             )
+
     def test_replication_state(self):
         """test replication state method"""
+        def getitem(name):
+            db_dict = {'_replicator': CouchDatabase(
+                mock_account,
+                '_replicator'
+            )}
+            return db_dict[name]
         mock_account = mock.Mock()
-        repl = ReplicatorDatabase(mock_account)
+        mock_account.__getitem__ = mock.Mock()
+        mock_account.__getitem__.side_effect = getitem
+        repl = Replicator(mock_account)
 
         mock_doc = mock.Mock()
         mock_doc.fetch = mock.Mock()
         mock_doc.get = mock.Mock()
         mock_doc.get.return_value = "STATE"
 
-        repl['replication_1'] = mock_doc
+        repl.database['replication_1'] = mock_doc
         self.assertEqual(repl.replication_state('replication_1'), 'STATE')
 
-        with mock.patch('cloudant.replicator.ReplicatorDatabase.__getitem__') as mock_gi:
+        with mock.patch('cloudant.database.CouchDatabase.__getitem__') as mock_gi:
```

Comment:
Yea, after you explained in detail the mock tests on Slack, it makes sense to change this at a later date.